src/load.jl

```
@@ -1,85 +1,71 @@
 
+    if excludeFolder !== nothing
+        exclude = push!(exclude, excludeFolder)
+    end
 
-    files_df = DataFrame(subject=[], ses=[], task=[], run=[], file=[], path=[])  # Initialize an empty DataFrame to hold results
-
-    # regular expression for additional information
-    # regex = r"sub-(.+)|_task-(.+)|_run-(.+)_eeg"
+    # Choose a specific folder in either ./ or ./derivatives
+    if specificFolder !== nothing
+        bidsPath = joinpath(bidsPath, specificFolder)
+    end
 
 
-    # Search for files matching file pattern
-    if specificFolder !== nothing
-        for (root, dirs, files) in walkdir(sPath)
-            for file in files
 
-                if sum(occursin.(file_pattern, file)) >= nPattern
+    files_df = DataFrame(subject=[], ses=[], task=[], run=[], file=[])  # Initialize an empty DataFrame to hold results
 
-                    get_info!(files_df, root, file)
-                end
+    list_all_eegpaths(path) = @cont begin
+        if isfile(path)
+            (any(endswith.(path, file_ending)) & all(occursin.(file_pattern, path))) && cont(path)
+        elseif isdir(path)
+            startswith(basename(path), ".") && return # skip all hidden files/ paths
+            if exclude !== nothing
+                basename(path) in (exclude...,) && return
             end
-        end
-
-        # When no specific folder is given look up whole Path    
-    else
-        for (root, dirs, files) in walkdir(bidsPath)
-            #filter(x->!startswith(x, '.'), files)
-            for file in files
-                if sum(occursin.(file_pattern, file)) >= nPattern &&
-                   (derivative && (exclude == "" || !any(occursin.(exclude, root))) ||
-                    (!derivative && !any(occursin.(exclude, root))))
-
-                    get_info!(files_df, root, file)
-                end
+            for file in readdir(path)
+                foreach(cont, list_all_eegpaths(joinpath(path, file)))
             end
         end
```

I would add an `else @throw "file not found $path"`

---

Changed this so there is an error if no files at all are found.